source/routing/raptor_api.cpp

```diff
@@ -856,6 +910,63 @@ parse_datetimes(RAPTOR& raptor,
     return datetimes;
 }
 
+pbnavitia::Response make_pt_response(RAPTOR &raptor,
+                                  const std::vector<type::EntryPoint> &origins,
+                                  const std::vector<type::EntryPoint> &destinations,
+                                  const uint64_t timestamp,
+                                  bool clockwise,
+                                  const type::AccessibiliteParams& accessibilite_params,
+                                  const std::vector<std::string>& forbidden,
+                                  const type::RTLevel rt_level,
+                                  const navitia::time_duration& transfer_penalty,
+                                  uint32_t max_duration,
+                                  uint32_t max_transfers,
+                                  bool show_codes,
+                                  uint32_t max_extra_second_pass){
+    log4cplus::Logger logger = log4cplus::Logger::getInstance(LOG4CPLUS_TEXT("logger"));
+    pbnavitia::Response response;
+    std::vector<bt::ptime> datetimes;
+    datetimes = parse_datetimes(raptor, {timestamp}, response, clockwise);
+    if(response.has_error() || response.response_type() == pbnavitia::DATE_OUT_OF_BOUNDS) {
+        return response;
+    }
+    auto datetime = datetimes.front();
+
+    routing::map_stop_point_duration departures;
+    routing::map_stop_point_duration arrivals;
+
+    for(const auto& origin: origins){
+        //todo check uri is valid
+        departures[SpIdx{*raptor.data.pt_data->stop_points_map[origin.uri]}] = navitia::seconds(origin.access_duration);
```

Comment:
isn't it a bit dangerous to acces the map like that with uri from outside the kraken (even if it was the previous kraken self that got them :wink: )
I would prefer some checks